sql/src/main/java/io/crate/execution/engine/window/RowNumberWindowFunction.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.execution.engine.window;
+
+import io.crate.metadata.BaseFunctionResolver;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionImplementation;
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.functions.params.FuncParams;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+
+import java.util.List;
+
+public class RowNumberWindowFunction {
+
+    private static final String NAME = "row_number";
+
+    static class RowNumberWindowFunctionImplementation implements WindowFunction {
```

Comment:
Any reason why we're not having the top-level class implement the `WindowFunction` directly?

Reply:
I was thinking of adding later the concrete implementations for `rank()` as `dense_rank()` which share the same signature as `row_number()` and then let the registration in the top-level class (which would be renamed).

What do you think?

Reply:
ok, fair comment, I'll change this accordingly.